app/models/concerns/profile_fields.rb

```diff
@@ -39,8 +39,8 @@ def canonical?
     end
 
     def rules_and_rule_groups
-      conflicts = relationships_for('conflicts')
-      requires = relationships_for('requires')
+      conflicts = map_rule_group_relationships('conflicts')
+      requires = map_rule_group_relationships('requires')
```

Comment:
Sometimes the solution is not to have an additional method, just to iterate :wink:
```suggestion
      conflicts, requires = %w[conflicts requires].map do |relationship|
        relationships_for(rules, rule_groups, relationship).each _with_object({}) do |rgr, relationships|
          relationships[rgr.left] ||= []
          relationships[rgr.left] << rgr.right
        end
      end
```